File: bandlog-api/src/main/java/net/effize/bandlog/domain/common/exception/ErrorCode.java
package net.effize.bandlog.domain.common.exception;

public class ErrorCode {
    public static ErrorCode BAD_REQUEST = ErrorCode.of("BAD_REQUEST");
    public static ErrorCode INTERNAL_SERVER_ERROR = ErrorCode.of("INTERNAL_SERVER_ERROR");
    private String value;

    private ErrorCode(String value) {
        this.value = value;
    }

    public static ErrorCode of(String value) {
        return new ErrorCode(value);
    }
}

File: src/main/kotlin/net/effize/bandlog/common/exception/ErrorCode.kt
package net.effize.bandlog.common.exception

@JvmInline
value class ErrorCode private constructor(val value: String) {
    companion object {
        val BAD_REQUEST = ErrorCode("BAD_REQUEST")
        val INTERNAL_SERVER_ERROR = ErrorCode("INTERNAL_SERVER_ERROR")

        fun of(value: String): ErrorCode = ErrorCode(value)
    }

    override fun toString(): String = value
}
